MSAL/src/MSALAccount.m

```diff
@@ -184,4 +203,46 @@ - (BOOL)isEqualToUser:(MSALAccount *)user
     return result;
 }
 
+#pragma mark - Tenant profiles
+
+- (NSArray<MSALTenantProfile *> *)loadAllTenantProfiles:(MSALPublicClientApplication *)application error:(NSError **)error
+{
+    if (!application)
+    {
+        MSAL_ERROR_PARAM(nil, MSALErrorInvalidParameter, @"application is a required parameter to load tenant profiles and must not be nil or empty.");
+        return nil;
+    }
+    
+    // if tenant profiles are already laoded, just return it
+    if (self.tenantProfiles) return self.tenantProfiles;
+    
+    MSALAccountsProvider *accountProvider = [[MSALAccountsProvider alloc] initWithTokenCache:application.tokenCache clientId:application.clientId];
+    
+    if (!accountProvider)
+    {
+        MSAL_ERROR_PARAM(nil, MSALErrorInternal, @"Failed to create account provider when loading all tenant profile!");
+        return nil;
+    }
+    
+    MSALAccount *accountWithTenatnProfiles = [accountProvider accountForHomeAccountId:self.homeAccountId.identifier error:error];
```

Comment:
Will there always be home account ID. What's about legacy accounts?

Reply:
This part of lazy loading is gone.